api/src/main/java/org/openmrs/PersonAttribute.java

```diff
@@ -194,13 +195,15 @@ public void setAttributeType(PersonAttributeType attributeType) {
 	 * @return the value
 	 */
 	public String getValue() {
-		return value;
+		return this.value;
 	}
 	
 	/**
 	 * @param value the value to set
 	 */
 	public void setValue(String value) {
+		super.setValue(value);
+		setValueReferenceInternal(value);
 		this.value = value;
 	}
```

Comment:
I realise this probably doesn't solve everything, but this is where we ultimately want to wind up:

```suggestion
	public void setValue(String value) {
		super.setValue(value);
	}
```